compiler/z/codegen/S390Instruction.hpp

```diff
@@ -2143,40 +2143,56 @@ class S390RILInstruction : public TR::Instruction
       isImmediateOffsetInBytesFlag    = 0x02
       };
 
-   /** Use to store immediate value in extended immediate instruction. */
-   int32_t  _sourceImmediate;
+   /** Use to store immediate value where the immediate is not used as a relative offset in address calculation.*/
+   uint32_t  _sourceImmediate;
 
    public:
 
    S390RILInstruction(TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register  *treg,
-                         uintptrj_t     tp,
+                         uint32_t           imm,
                          TR::CodeGenerator *cg)
-      : TR::Instruction(op, n, cg), _targetSnippet(NULL), _targetSymbol(NULL), _flagsRIL(0), _mask(0xffffffff),  _targetLabel(NULL),_symbolReference(NULL)
+      : TR::Instruction(op, n, cg), _targetPtr(NULL), _targetSnippet(NULL), _targetSymbol(NULL), _flagsRIL(0), _mask(0xffffffff),  _targetLabel(NULL), _symbolReference(NULL), _sourceImmediate(imm)
       {
+      TR_ASSERT_FATAL(getOpCode().isExtendedImmediate(), "Incorrect use of RIL constructor");
+      checkRegForGPR0Disable(op,treg);
+      if (!getOpCode().setsOperand1())
+         useSourceRegister(treg);
+      else
+         useTargetRegister(treg);
+      }
 
+   S390RILInstruction(TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register  *treg,
+                         int32_t            imm,
+                         TR::CodeGenerator *cg)
+      : TR::Instruction(op, n, cg), _targetPtr(NULL), _targetSnippet(NULL), _targetSymbol(NULL), _flagsRIL(0), _mask(0xffffffff),  _targetLabel(NULL), _symbolReference(NULL), _sourceImmediate(imm)
+      {
+      TR_ASSERT_FATAL(getOpCode().isExtendedImmediate(), "Incorrect use of RIL constructor");
```

Comment:
Same as above, and all other places in this commit (I'll stop commenting on this).